tests/unittests/test_util.py

```diff
@@ -2368,103 +2367,95 @@ def test_output_logs_parsed_when_teeing_files_and_rotated(self):
+class TestMultiLog:
+    def _createConsole(self):
+        os.mkdir("/dev")
+        open("/dev/console", "w").close()
 
-    def setUp(self):
-        super(TestMultiLog, self).setUp()
-        self.root = tempfile.mkdtemp()
-        self.addCleanup(shutil.rmtree, self.root)
-        self.patchOS(self.root)
-        self.patchUtils(self.root)
-        self.patchOpen(self.root)
-        self.stdout = io.StringIO()
-        self.stderr = io.StringIO()
-        self.patchStdoutAndStderr(self.stdout, self.stderr)
-
-    def test_stderr_used_by_default(self):
+    def test_stderr_used_by_default(self, capsys):
         logged_string = "test stderr output"
         log_util.multi_log(logged_string)
-        self.assertEqual(logged_string, self.stderr.getvalue())
+        assert logged_string == capsys.readouterr().err
 
-    def test_stderr_not_used_if_false(self):
+    def test_stderr_not_used_if_false(self, capsys):
         log_util.multi_log("should not see this", stderr=False)
-        self.assertEqual("", self.stderr.getvalue())
+        assert "" == capsys.readouterr().err
 
     def test_logs_go_to_console_by_default(self):
-        self._createConsole(self.root)
+        self._createConsole()
         logged_string = "something very important"
         log_util.multi_log(logged_string)
         with open("/dev/console") as f:
-            self.assertEqual(logged_string, f.read())
+            assert logged_string == f.read()
 
-    def test_logs_dont_go_to_stdout_if_console_exists(self):
-        self._createConsole(self.root)
+    def test_logs_dont_go_to_stdout_if_console_exists(self, capsys):
+        self._createConsole()
         log_util.multi_log("something")
-        self.assertEqual("", self.stdout.getvalue())
+        assert "" == capsys.readouterr().out
 
-    def test_logs_go_to_stdout_if_console_does_not_exist(self):
+    def test_logs_go_to_stdout_if_console_does_not_exist(self, capsys):
         logged_string = "something very important"
         log_util.multi_log(logged_string)
-        self.assertEqual(logged_string, self.stdout.getvalue())
+        assert "something very important" == capsys.readouterr().out
 
-    def test_logs_dont_go_to_stdout_if_fallback_to_stdout_is_false(self):
+    def test_logs_dont_go_to_stdout_if_fallback_to_stdout_is_false(
+        self, capsys
+    ):
         log_util.multi_log("something", fallback_to_stdout=False)
-        self.assertEqual("", self.stdout.getvalue())
+        assert "" == capsys.readouterr().out
 
-    @mock.patch(
-        "cloudinit.log.log_util.write_to_console",
-        mock.Mock(side_effect=OSError("Failed to write to console")),
-    )
     def test_logs_go_to_stdout_if_writing_to_console_fails_and_fallback_true(
-        self,
+        self, mocker, capsys
     ):
-        self._createConsole(self.root)
+        mocker.patch(
+            "cloudinit.log.log_util.write_to_console",
+            mock.Mock(side_effect=OSError("Failed to write to console")),
```

Comment:
I don't think we need to wrap the side_effect in a mock as we're already performing that mocker.patch which accepts the side_effect arg right?
```suggestion
           side_effect=OSError("Failed to write to console"),
```